src/main/kotlin/blackjack/model/BlackjackGame.kt

```diff
@@ -0,0 +1,36 @@
+package blackjack.model
+
+import blackjack.states.Running
+
+class BlackjackGame(
+    private val dealer: Dealer,
+    private val players: List<Player>,
+    private val askForCard: (Player) -> Boolean,
+    private val output: (Player) -> Unit,
+) {
+    val cardDeck = CardDeck()
+
+    init {
+        setUpTable()
+    }
+
+    fun setUpTable() {
+        players.forEach { player ->
+            repeat(2) {
+                player.state = player.state.draw(cardDeck.drawCard())
+            }
+        }
+        repeat(2) { dealer.state = dealer.state.draw(cardDeck.drawCard()) }
+    }
+
+    fun play() {
+        dealer.playTurn(cardDeck)
+        players.forEach {
+            while (it.state is Running && askForCard(it)) {
```

Comment:
Instead of using `is`, how about leveraging polymorphism? It can lead to cleaner and more maintainable code. 🙂